wandb/sdk/artifacts/artifact.py

```diff
@@ -814,102 +760,91 @@ def wait(self, timeout: int | None = None) -> Artifact:
         return self
 
     def _populate_after_save(self, artifact_id: str) -> None:
-        fields = InternalApi().server_artifact_introspection()
-
-        supports_ttl = "ttlIsInherited" in fields
-        ttl_duration_seconds = "ttlDurationSeconds" if supports_ttl else ""
-        ttl_is_inherited = "ttlIsInherited" if supports_ttl else ""
-
-        supports_tags = "tags" in fields
-        tags = "tags {name}" if supports_tags else ""
-
-        query_template = f"""
+        query = gql(
+            f"""
             query ArtifactByIDShort($id: ID!) {{
                 artifact(id: $id) {{
-                    artifactSequence {{
-                        project {{
-                            entityName
-                            name
-                        }}
-                        name
-                    }}
-                    versionIndex
-                    {ttl_duration_seconds}
-                    {ttl_is_inherited}
-                    aliases {{
-                        artifactCollection {{
-                            project {{
-                                entityName
-                                name
-                            }}
-                            name
-                        }}
-                        alias
-                    }}
-                    {tags!s}
-                    state
+                    ...ArtifactFragment
                     currentManifest {{
                         file {{
                             directUrl
                         }}
                     }}
-                    commitHash
-                    fileCount
-                    createdAt
-                    updatedAt
                 }}
             }}
-        """
-
-        query = gql(query_template)
-
-        assert self._client is not None
-        response = self._client.execute(
-            query,
-            variable_values={"id": artifact_id},
+            {self._get_gql_artifact_fragment()}
+            """
         )
+        assert self._client is not None  # Appease mypy.
```

Comment:
suggestion(optional): optional since I realize this is existing code, but -- 

Since the `assert` is mainly to satisfy type checks, I think it'd be better if we could drop the `assert` and go "all or nothing" on the typing here.  By which I mean:

"all": runtime validation (without assert).  I.e., raise a more typical `RuntimeError` if `self._client` really is None (since there isn't supposed to be a codepath through which this actually happens)
```suggestion
        if self._client is None:
            raise RuntimeError("Client missing: unable to populate artifact")
```

"...or nothing": appease `mypy` use `typing.cast`, which tells `mypy` what type to assume (but intentionally does pretty much nothing at runtime)
```suggestion
        self._client = cast(Client, self._client)  # Appease mypy.
```